Dockerfile-cuda

```diff
@@ -1,4 +1,4 @@
-FROM nvidia/cuda:12.2.0-devel-ubuntu22.04 AS base-builder
+FROM nvidia/cuda:12.9.0-devel-ubuntu22.04 AS base-builder
```

Comment:
Is bumping CUDA required? As it might eventually be a breaking change for instances running on older versions of NVIDIA as 12.2, 12.4 and 12.8; besides that everything LGTM